rotkehlchen/chain/ethereum/modules/thegraph/balances.py

```diff
@@ -0,0 +1,118 @@
+from collections import defaultdict
+import logging
+from typing import TYPE_CHECKING, NamedTuple, Optional
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.accounting.structures.evm_event import EvmProduct
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.interfaces.balances import ProtocolWithBalance, ProtocolWithGauges
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.chain.evm.tokens import get_chunk_size_call_order
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.assets import A_CVX, A_GRT
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+
+from .constants import CPT_THEGRAPH
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.structures.evm_event import EvmEvent
+    from rotkehlchen.chain.ethereum.interfaces.balances import BalancesType
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import CHAIN_IDS_WITH_BALANCE_PROTOCOLS
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class DelegationData(NamedTuple):
+    shares: FVal  # Shares owned by a delegator in the pool
+    tokensLocked: FVal  # Tokens locked for undelegation
+    tokensLockedUntil: FVal  # Block when locked tokens can be withdrawn
+
+class DelegationPoolData(NamedTuple):
+    cooldownBlocks: FVal  # Blocks to wait before updating parameters
+    indexingRewardCut: FVal  # in PPM
+    queryFeeCut: FVal  # in PPM
+    updatedAtBlock: FVal  # Block when the pool was last updated
```

Comment:
if you don't need all this information for the balances don't store everything, just what you need